### PR TITLE
Fix mkdirp promise definitions by declaring namespace

### PR DESCRIPTION
**Pull request by Voles**
Declare namespace to prevent 'resolves to a non-module entity' error when importing.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by Voles** [comment]
@pluma you might be interested in this PR

**Comment by pluma** [comment]
@Voles can you elaborate on why this is necessary? Unless something has changed, mkdirp-promise does not export an ES module and I don't see the warning when using `import mkdirp = require("mkdirp-promise")`, which is how you import CommonJS modules.

**Comment by pluma** [comment]
Specifically the test file uses that and doesn't produce any warnings AFAICT: mkdirp-promise-tests.ts

**Comment by pluma** [comment]
If other libraries are doing the same, I see no reason to reject this. However as I understand it the canonical way to import CJS modules is using the `import x = require('x')` syntax: https://www.typescriptlang.org/docs/handbook/modules.html#import (scroll down to "`export =` and `import = require()`".

**Review by pluma — APPROVED** [review]

**Comment by pluma** [comment]
IOW I think this change disagrees with what the TS docs say, but it seems to be consistent with established de facto practice so I would regard the TS docs as irrelevant and therefore approve of this change.

I'd like a @DefinitelyTyped maintainer to provide a second opinion tho.

**Comment by Voles** [comment]
@pluma thanks again for your feedback!

I have been reading more about this and you are right. This change is not the way to go. I will close this PR.

**For future reference**
[p-queue](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/p-queue/index.d.ts) is one of the other packages where I've seen this approach.

To import CJS modules, one should indeed use `import x = require('x')`. However, since TypeScript 2.7 there's also the flag [`--esmoduleInterop`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#support-for-import-d-from-cjs-form-commonjs-modules-with---esmoduleinterop) which allows you to use the `import x from 'x'` syntax.

However, this is not enabled by default (hence my struggle and this PR).

**Pull request closed without merging**
